add host: wording fixes

andreasn opened the pull request:
Shweta Jalgaonkar did a review of the text strings used in the add
host dialog, and these are the changes she recommended.

ShwetaNaresh commented:
Thanks, Andreas!

marusak commented on pkg/lib/machine-add.html:
```diff
@@ -1,17 +1,16 @@
 <div class="modal-body">
     <form class="ct-form">
-      <label translate="yes" class="control-label">Address</label>
+      <label translate="yes" class="control-label">IP address or hostname</label>
```
Won't this make the UI too sparse? (that the label will take half of the dialog)
(sure I should just build it and see...)

martinpitt replied:
This could also become a tooltip? Honestly, "address" is a very common concept in networking, and pretty much everywhere you can specify a host name or IPv4/IPv6 address.

marusak commented on pkg/lib/machine-add.html:
```diff
@@ -1,17 +1,16 @@
     <button class="close" data-dismiss="modal"
         aria-hidden="true">&times;</button>
-    <h4 class="modal-title" translate="yes">Add a New Host</h4>
+    <h4 class="modal-title" translate="yes">Add new host</h4>
```
This is gonna conflict with https://github.com/cockpit-project/cockpit/pull/14574 which is going to land very soon.

andreasn replied:
I'm happy to wait for that to land first

martinpitt commented:
This needs a rebase now, the big Title Case PR landed.

andreasn commented:
I have two different solutions to address the concern @marusak has with the long label.

1. Keep the string @ShwetaNaresh recommend, but move to labels-on-top [Patternfly guidelines](https://www.patternfly.org/v4/documentation/react/components/form)
![cockpit-labels-on-top](https://user-images.githubusercontent.com/209895/93063207-3391fa00-f676-11ea-9979-afbc871a3eea.png)

2. Keep the label address as in master, but use the string @ShwetaNaresh recommend as a helper text. [Patternfly guidelines](https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/text-input)
![cockpit-helper-text](https://user-images.githubusercontent.com/209895/93063212-342a9080-f676-11ea-8210-2433dff29627.png)

Leaning slightly towards 2 myself.

andreasn commented:
I had missed that @ShwetaNaresh also recommended to add a helper text at the top of the dialog:
`Specify the IP address or hostname and the administrative user account name for the host that you want to add.`

Will add that as well.

martinpitt commented:
@andreasn: This isn't complete -- the address can also contain an optional port. But it can even be an arbitrary alias name that the user configured in ~/.ssh/config. So perhaps we should really call it "SSH destination"? That's how `man ssh` calls these.

andreasn commented:
@ShwetaNaresh can you help me with a text that incorporates the port and alias that @martinpitt mentioned above?

ShwetaNaresh commented:
To address this case, do the following suggestions hold good?

Label: Rename the label to 'Host details'

Helper text at the top of the dialogue:
Specify the host details and the administrative user account name for the
host that you want to add.
For host details, either specify the IP address, hostname, alias name, or a
unique resource identifier for SSH destination.

Regards

Shweta


On Mon, Sep 14, 2020 at 11:25 AM Martin Pitt <notifications@github.com>
wrote:

> @andreasn <https://github.com/andreasn>: This isn't complete -- the
> address can also contain an optional port. But it can even be an arbitrary
> alias name that the user configured in ~/.ssh/config. So perhaps we should
> really call it "SSH destination"? That's how man ssh calls these.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockpit-project/cockpit/pull/14601#issuecomment-691934938>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMBFMT2Y5IQ5Z7PXIOHRKN3SFXOPXANCNFSM4RE6OHHQ>
> .
>

andreasn commented:
Updated version, with new strings.
![Screenshot from 2020-09-18 16-10-02](https://user-images.githubusercontent.com/209895/93607317-74b54180-f9c9-11ea-8d77-4476b0a70f25.png)

ShwetaNaresh commented:
Hi Andreas,

Just a minor update to the label please. It should read 'Host details'.
Currently, it is mentioned as 'Host detail'.

Thanks!
Shweta

On Fri, Sep 18, 2020, 4:10 PM andreasn <notifications@github.com> wrote:

> Updated version, with new strings.
> [image: Screenshot from 2020-09-18 16-10-02]
> <https://user-images.githubusercontent.com/209895/93607317-74b54180-f9c9-11ea-8d77-4476b0a70f25.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockpit-project/cockpit/pull/14601#issuecomment-694891589>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMBFMT63EWQ67H2VBI3YDI3SGNS6TANCNFSM4RE6OHHQ>
> .
>

andreasn commented:
> Hi Andreas, Just a minor update to the label please. It should read 'Host details'. Currently, it is mentioned as 'Host detail'. Thanks! Shweta
> […](#)
> On Fri, Sep 18, 2020, 4:10 PM andreasn ***@***.***> wrote: Updated version, with new strings. [image: Screenshot from 2020-09-18 16-10-02] <https://user-images.githubusercontent.com/209895/93607317-74b54180-f9c9-11ea-8d77-4476b0a70f25.png> — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#14601 (comment)](https://github.com/cockpit-project/cockpit/pull/14601#issuecomment-694891589)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AMBFMT63EWQ67H2VBI3YDI3SGNS6TANCNFSM4RE6OHHQ> .

Fixed!

marusak commented:
Hmm, I feel like we bike shed here way too much, but I cannot help myself and not mention that `Host details` does not indicate at all to me that it should be some name/address to which I want to connect. I am that kind of user that does not read helps, tooltips and manuals but tries stuff and hopes for the best. So in this case I would have a hard time - under `Host details` I would write probably some description of the server as I would think it is most likely some label that will show up somewhere in the UI to distinguish servers. Sure there is that help, but if it is longer than half a line I am not reading it until I try 5 things and still cannot connect.
Those were just my 2c, but I feel like combination 'Address' + 'User name' is much easier to understand as it is the usual pattern. If address would indicate it must be a IP address then `Name`, `Host`, `Destination` are some ideas that make sense to me.

martinpitt commented:
Agree to @mmarusak that "host details" is very confusing -- "address" is an established term, but this is exactly about *not* providing details, but a more abstract name -- the details are in the ssh config. So let's not break translations and make the string worse at the same time.

ShwetaNaresh commented:
Thank you, Matej and Martin!

I guess this brings us back to 'Hostname or IP address' as the label.

From a user perspective, what would be the preferred input?
I presume that hostname or IP address is a common input that users would
have, and so it would be good to have a related label.

Just 'Address' as the label seems incomplete. It does not seem to
completely relate to any of the possible user inputs.
For users who prefer to provide SSH destination, we can include the
information in the helper text. Again, presuming there would be less number
of users who would provide this input, I guess, a line in the helper text
should help them.

In this case, the helper text can be on the following lines:

++++++++
Specify the hostname or IP address and the administrative user account name
for the host that you want to add.
In place of hostname or IP address, you can choose to provide an alias name
or a unique resource identifier for the SSH destination.
++++++++

Please let me know if this holds good?

Regards

Shweta

On Tue, Sep 22, 2020 at 8:47 AM Martin Pitt <notifications@github.com>
wrote:

> Agree to @mmarusak <https://github.com/mmarusak> that "host details" is
> very confusing -- "address" is an established term, but this is exactly
> about *not* providing details, but a more abstract name -- the details
> are in the ssh config. So let's not break translations and make the string
> worse at the same time.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockpit-project/cockpit/pull/14601#issuecomment-696540354>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMBFMT666J4AX5I4RPZWEKTSHBCBRANCNFSM4RE6OHHQ>
> .
>

marusak commented:
> I guess this brings us back to 'Hostname or IP address' as the label.

Yeah, I guess that would work. But at the same time I think that we should not overthinking it and just putting there `Host` could work. That of course does not explain what values are accepted but for that we have the helper text. And I believe that the common pattern would just be intuitive - Pick **host** you want to connect to (if you are used to doing `ssh foobar` put there `foobar` and if `ssh 1.1.1.1` then just put there `1.1.1.1`

martinpitt commented:
> I think that we should not overthinking it and just putting there `Host` could work.

I actually like that because that's *exactly* the keyword in ~/.ssh/config, in `man ssh_config` etc.

ShwetaNaresh commented:
Sure, Matej/Martin.

Would be good to modify the helper text a bit:

++++++++
Specify the host and the administrative user account name for the host that
you want to add.
For the host, either specify the hostname, IP address, an alias name or a
unique resource identifier for the SSH destination.
++++++++

Regards

Shweta

On Thu, Sep 24, 2020 at 12:55 PM Martin Pitt <notifications@github.com>
wrote:

> I think that we should not overthinking it and just putting there Host
> could work.
>
> I actually like that because that's *exactly* the keyword in
> ~/.ssh/config, in man ssh_config etc.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockpit-project/cockpit/pull/14601#issuecomment-698270909>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMBFMT5T3MHFPAPOLNYYUDTSHMQTBANCNFSM4RE6OHHQ>
> .
>

marusak commented:
> Specify the host and the administrative user account name for the host that
you want to add.
For the host, either specify the hostname, IP address, an alias name or a
unique resource identifier for the SSH destination.

Sure :) Not convinced that we need "administrative" there though.

andreasn commented:
> Not convinced that we need "administrative" there though.

Does it work with accounts that are not in wheel?

marusak commented:
> Does it work with accounts that are not in wheel?

yes

ShwetaNaresh commented:
Thanks, Matej.

I presumed we need admin access to perform any administrative tasks on the
system, especially if the user opts for auto-login.

Is that not the case?

Regards

Shweta


On Mon, Sep 28, 2020, 11:42 AM Matej Marusak <notifications@github.com>
wrote:

> Does it work with accounts that are not in wheel?
>
> yes
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockpit-project/cockpit/pull/14601#issuecomment-699899870>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMBFMT3ST5MDKNQ72KW5DPDSIBLCBANCNFSM4RE6OHHQ>
> .
>

marusak commented:
> I presumed we need admin access to perform any administrative tasks

that is correct, but you don't need administrative access to log into server (nor to add server into the switcher).

ShwetaNaresh commented:
So, if a user specifies a non-admin user account while he adds the host and
chooses auto-login for next sessions, will the web console not block him to
perform any tasks on the system?
Will the web console ask for admin user credentials later?

Regards

Shweta

On Mon, Sep 28, 2020, 12:17 PM Matej Marusak <notifications@github.com>
wrote:

> I presumed we need admin access to perform any administrative tasks
>
> that is correct, but you don't need administrative access to log into
> server (nor to add server into the switcher).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockpit-project/cockpit/pull/14601#issuecomment-699917504>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMBFMT6YLBHM2L4UC7GWVTTSIBPDDANCNFSM4RE6OHHQ>
> .
>

marusak commented:
> chooses auto-login for next sessions

Where does one chooses that?

> will the web console not block him to perform any tasks on the system?
Will the web console ask for admin user credentials later?

I am not sure if I follow what is the issue. A non admin can log into Cockpit it is just about what such user can and cannot do. We disable buttons and don't show things that this user does not have permissions for.

ShwetaNaresh commented:
Discussed with Matej.

Here's the modified text:

+++++++

Specify the host and the login user account for the host that you want to
add.

For the host, either specify the hostname, IP address, an alias name or a
unique resource identifier for the SSH destination.
+++++

Does it sound good? Please let me know the changes required, if any.

Regards

Shweta

On Mon, Sep 28, 2020 at 1:49 PM Matej Marusak <notifications@github.com>
wrote:

> chooses auto-login for next sessions
>
> Where does one chooses that?
>
> will the web console not block him to perform any tasks on the system?
> Will the web console ask for admin user credentials later?
>
> I am not sure if I follow what is the issue. A non admin can log into
> Cockpit it is just about what such user can and cannot do. We disable
> buttons and don't show things that this user does not have permissions for.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockpit-project/cockpit/pull/14601#issuecomment-699958030>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMBFMT52THBOZN6SKOQYS3TSIBZ57ANCNFSM4RE6OHHQ>
> .
>

andreasn commented:
Pushed the suggested fixes. Now looks like this:
![image](https://user-images.githubusercontent.com/209895/94945828-8de6e380-04db-11eb-9b62-412ba6da62bd.png)

marusak reviewed (dismissed):
Thanks, now I don't have any more nacks :)

martinpitt commented on pkg/lib/machine-add.html:
```diff
@@ -1,12 +1,14 @@
 </div>
 
 <div class="modal-body">
+    <p>Specify the host and the login user account for the host that you want to add.<br/>
+    For the host, either specify the hostname, IP address, an alias name or a unique resource identifier for the SSH destination.</p><br/>
```
This is a bit awkward for translations. The `<br/>` is part of the text to be translated, and there's extra line breaks and spaces. Also, these are missing `translate` HTML attributes, so that they get marked for translation.

I suggest splitting the sentences and wrapping them into two `<span>`s, so that the halves get translated separately, without any surrounding whitespace or HTML tags.

martinpitt reviewed (approved):
Cheers!
